### DISCUSSION TITLE
Splitting DoH, API and Metrics endpoints

### DISCUSSION THREAD

**Discussion by xintron** [0xERR0R/blocky#1564] [General]
After reviewing the documentation, it appears that enabling DoH for public devices also opens the API and metrics to the public. This issue can be mitigated by running a proxy in front that only exposes /dns-query, but this adds an extra, seemingly unnecessary, step.

Wouldn't it be more practical to have separate address and port configurations for DoH (`0.0.0.0:443`) and for API/pprof/metrics (`127.0.0.1:8443`)? This way, users can easily control what they want to expose without needing to use traefik/nginx to manage access.

**Comment by ThinkChaos** [comment]
Hi, yes we agree :)

See issue #1206 and PR #1427
